executor.go

```diff
@@ -278,15 +279,26 @@ func executeFields(p ExecuteFieldsParams) *Result {
 		p.Fields = map[string][]*ast.Field{}
 	}
 
+	var wg sync.WaitGroup
+	wg.Add(len(p.Fields))
+
+	var resultsMutex sync.Mutex
 	finalResults := map[string]interface{}{}
 	for responseName, fieldASTs := range p.Fields {
-		resolved, state := resolveField(p.ExecutionContext, p.ParentType, p.Source, fieldASTs)
-		if state.hasNoFieldDefs {
-			continue
-		}
-		finalResults[responseName] = resolved
+		go func(responseName string, fieldASTs []*ast.Field) {
```

Comment:
I see what you're getting at now, but I feel like it's reasonable to assume that someone that handles 100s of requests per second would allocate the necessary resources to do so. Also that someone with such a large schema would have their own optimizations for hitting external resources.

On the other hand, perhaps just spawning a goroutine per resolve is a little lazy. Another solution could be to only spawn a goroutine when a `func() interface{}` is returned by the resolve function.